traffic_ops/testing/api/v4/profile_parameters_test.go

```diff
@@ -144,35 +190,274 @@ func GetTestProfileParameters(t *testing.T) {
+	}
+
+	profileID1 := profileResp1.Response[0].ID
+	parameterID1 := paramResp1.Response[0].ID
+	profileID2 := profileResp2.Response[0].ID
+	parameterID2 := paramResp2.Response[0].ID
+
+	DeleteTestProfileParameter(t, profileID1, parameterID1)
+	DeleteTestProfileParameter(t, profileID2, parameterID2)
+
+	pp := tc.ProfileParameterCreationRequest{
+		ProfileID:   profileID1,
+		ParameterID: parameterID1,
+	}
+	pp2 := tc.ProfileParameterCreationRequest{
+		ProfileID:   profileID2,
+		ParameterID: parameterID2,
+	}
+
+	ppSlice := []tc.ProfileParameterCreationRequest{
+		pp,
+		pp2,
+	}
+	_, _, err = TOSession.CreateMultipleProfileParameters(ppSlice, client.RequestOptions{})
+	if err != nil {
+		t.Errorf("could not CREATE profile parameters: %v", err)
+	}
+}
+
+func CreateProfileWithMultipleParameters(t *testing.T) {
+	if len(testData.Parameters) < 2 || len(testData.Profiles) < 2 {
+		t.Fatal("Need at least two Profile and two Parameter to test associating a Parameter with a Profile")
+	}
+	firstProfile1 := testData.Profiles[0]
+	opts := client.NewRequestOptions()
+	opts.QueryParameters.Set("name", firstProfile1.Name)
+	profileResp1, _, err := TOSession.GetProfiles(opts)
+	if err != nil {
+		t.Errorf("cannot GET Profile by name: %v - %v", firstProfile1.Name, err)
+	}
+	if len(profileResp1.Response) != 1 {
+		t.Fatalf("Expected exactly one Profile to exist with name '%s', found: %d", firstProfile1.Name, len(profileResp1.Response))
+	}
+
+	firstProfile2 := testData.Profiles[1]
+	opts.QueryParameters.Set("name", firstProfile2.Name)
+	profileResp2, _, err := TOSession.GetProfiles(opts)
+	if err != nil {
+		t.Errorf("cannot GET Profile by name: %v - %v", firstProfile2.Name, err)
+	}
+	if len(profileResp2.Response) != 1 {
+		t.Fatalf("Expected exactly one Profile to exist with name '%s', found: %d", firstProfile2.Name, len(profileResp2.Response))
+	}
+
+	firstParameter1 := testData.Parameters[0]
+	opts.QueryParameters.Set("name", firstParameter1.Name)
+	paramResp1, _, err := TOSession.GetParameters(opts)
+	if err != nil {
+		t.Errorf("cannot get Parameter by name '%s': %v - alerts: %+v", firstParameter1.Name, err, paramResp1.Alerts)
+	}
+	if len(paramResp1.Response) < 1 {
+		t.Fatalf("Expected at least one Parameter to exist with name '%s', found: %d", firstParameter1.Name, len(paramResp1.Response))
+	}
+
+	firstParameter2 := testData.Parameters[1]
+	opts.QueryParameters.Set("name", firstParameter2.Name)
+	paramResp2, _, err := TOSession.GetParameters(opts)
+	if err != nil {
+		t.Errorf("cannot get Parameter by name '%s': %v - alerts: %+v", firstParameter2.Name, err, paramResp2.Alerts)
+	}
+	if len(paramResp2.Response) < 1 {
+		t.Fatalf("Expected at least one Parameter to exist with name '%s', found: %d", firstParameter2.Name, len(paramResp2.Response))
+	}
+
+	profileID1 := profileResp1.Response[0].ID
+	parameterID1 := paramResp1.Response[0].ID
+	profileID2 := profileResp2.Response[0].ID
```

Comment:
If the first Profile is the only one you're actually using for the test of assigning multiple Parameters to a Profile, why even bother fetching a second Profile?

Reply:
Removed that line.